chain/network/src/peer_manager/network_state/mod.rs

```diff
@@ -92,10 +85,11 @@ pub(crate) struct NetworkState {
     /// WARNING: actix actors can be spawned only when actix::System::current() is set.
     /// DO NOT spawn actors from a task on this runtime.
     runtime: Runtime,
+    /// Address for the peer_manager_actor, used to spawn reconnect attempts.
+    /// Can be None in tests.
+    pub peer_manager_actor: Option<Addr<PeerManagerActor>>,
```

Comment:
Don't introduce a cyclical dependency on PeerManagerActor (avoid cyclical dependencies in general). Spawn stuff on NetworkState::runtime instead.

Reply:
you can move some logic from PeerManagerActor to NetworkState if needed.

Reply:
As discussed offline we can't spawn PeerActors on the NetworkState runtime. Instead, the plan is to have PeerManagerActor spawn reconnect loops upon polling the connection store.